lib/Serialiser/Ntriples.php

```diff
@@ -49,27 +49,143 @@
  */
 class Ntriples extends Serialiser
 {
-    private $escChars = [];   // Character encoding cache
 
     /**
-     * @ignore
+     * Escapes a string literal according to the N-Triples specification.
+     *
+     * The sequence with which the characters are replaced is important.
+     *
+     * The sequence of replacements is:
+     * - Backslash (\) - Escape character. It is escaped first so that we do not
+     *  double escape other sequences that contain a backslash.
+     * - Control characters (0-31) and DEL (127) - except \t, \n, \r and \".
+     *  These are replaced with their unicode representation using a normal
+     *  str_replace because they break preg_replace_callback.
+     * - 8-byte characters - These are replaced with their unicode
+     * representation using a preg_replace_callback. They are replaced first so
+     * that they are not replaced as multiple characters down below.
+     * - 4-byte characters - These are replaced with their unicode
+     * representation using a preg_replace_callback. They are replaced next so
+     * that they are not replaced as multiple characters down below.
+     * - Characters from 127 to 255 - These are replaced with their unicode
+     * representation using a normal str_replace because fail to match in
+     * preg_replace_callback. This takes place here because otherwise, they
+     * might be confused with multibyte characters if it takes place before the
+     * two steps above. And it takes place after the above two steps, in order
+     * to avoid replacing them as multiple characters.
+     * - Characters from 32 to 126 - These are not escaped as they are printable
+     * characters. However, the double quote (\") is escaped above as it is a
+     * special character in N-Triples.
```

Comment:
Let's arrange the indentation for bullet points. The lines should be indented same at the position where the 1st line's text starts:

```
- abc
  xyz
```